examples/pygeosxExamples/parametricSurfaceMapping/run_parametric_mesh_problem.py

```diff
@@ -0,0 +1,61 @@
+
+import sys
+import numpy as np
+from mpi4py import MPI
+import pygeosx
+from pygeosx_tools import wrapper
+
+
+def surface_mapping(x):
+    """
+    Function to deform the mesh to match some surface topography
+
+    Args:
+        x (np.ndarray) the object locations
+
+    Returns:
+        np.ndarray: new coordinates
+    """
+    z_transition = 0.5   # Depth of the zone to be deformed
+    z_surf = 1.0         # Average location of the model surface
+
+    R = x[:, 0]**2 + x[:, 1]**2
+    delta_surf = 0.1 * np.sin(2.0 * np.pi * R / np.amax(R))
+    delta_z = delta_surf * np.maximum(x[:, 2] - z_surf + z_transition, 0.0) / z_transition
+    return x[:, 2] + delta_z
+
+
+def run_problem():
+    """
+    Run the GEOSX problem
+    """
+    # Initialize the code
+    comm = MPI.COMM_WORLD
+    rank = comm.Get_rank()
+    problem = pygeosx.initialize(rank, sys.argv)
+
+    # Rather than specifying the wrapper paths explicitly,
+    # search for them using a set of filters
+    location_keys = []
+    location_keys.append(wrapper.get_matching_wrapper_path(problem, ['ReferencePosition']))
+    location_keys.append(wrapper.get_matching_wrapper_path(problem, ['Region2', 'elementCenter']))
+
+    # Map the node, element locations
+    print('Warning: this mapping will change the mesh, but for things to work correctly')
+    print('         we would need to re-calculate the geometric parameters, shape functions')
```

Comment:
What would be necessary to get this to work correctly? Obviously we could add in a call to re-calculate things, but could we add in another step between `initialize` and `apply_initial_conditions` so that you only calculate these things once?

Reply:
Sure, to get this working we'd one of the following:

- A callback to python somewhere after the mesh is created, but before the shape functions are calculated
- A method to trigger shape function updates

Reply:
Cool, is there any reason we'd want option 2 over 1? Are you ever going to want to recalculate stuff mid run?